rest/utilities_testing.go

```diff
@@ -62,6 +62,7 @@ type RestTesterConfig struct {
 	persistentConfig                bool
 	groupID                         *string
 	createScopesAndCollections      bool // If true, will automatically create any defined scopes and collections on startup.
+	useLeakyTBForDB                 bool // If true, the leaky test bucket will be used on the database. Useful for using a LeakyBucket on the DB context for integration tests.
```

Comment:
Can this be inferred if `RestTesterConfig.TestBucket` is non-nil? (is there a situation where you _wouldn't_ want to use the leaky test bucket if you were passing one in?)

Reply:
Did try that way at first, however some tests actually rely on the DB generating the bucket and therefore failing. For example, `TestCollectionsBasicIndexQuery` which relies on the index not being created on the default collection however it does get created on the default collection when TB is passed in to the database context.